internal/cmd/connect/command_cluster_delete.go

```diff
@@ -41,31 +37,57 @@ func (c *clusterCommand) newDeleteCommand() *cobra.Command {
 }
 
 func (c *clusterCommand) delete(cmd *cobra.Command, args []string) error {
-	clusterId := args[0]
-	kafkaCluster, err := c.Context.GetKafkaClusterForCommand()
+	environmentId, err := c.Context.EnvironmentId()
 	if err != nil {
 		return err
 	}
 
-	environmentId, err := c.Context.EnvironmentId()
+	kafkaCluster, err := c.Context.GetKafkaClusterForCommand()
 	if err != nil {
 		return err
 	}
 
-	connector, err := c.V2Client.GetConnectorExpansionById(clusterId, environmentId, kafkaCluster.ID)
-	if err != nil {
+	connectorIdToName := make(map[string]string)
+	if err := c.confirmDeletion(cmd, environmentId, kafkaCluster.ID, args, connectorIdToName); err != nil {
 		return err
 	}
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmMsg, resource.Connector, clusterId, connector.Info.GetName())
-	if _, err := form.ConfirmDeletion(cmd, promptMsg, connector.Info.GetName()); err != nil {
+	deleteFunc := func(id string) error {
+		if _, err := c.V2Client.DeleteConnector(connectorIdToName[id], environmentId, kafkaCluster.ID); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	deleted, err := resource.Delete(args, deleteFunc, nil)
+	resource.PrintDeleteSuccessMsg(deleted, resource.Connector)
+
+	return err
+}
+
+func (c *clusterCommand) confirmDeletion(cmd *cobra.Command, environmentId, kafkaClusterId string, args []string, connectorIdToName map[string]string) error {
+	describeFunc := func(id string) error {
+		connector, err := c.V2Client.GetConnectorExpansionById(id, environmentId, kafkaClusterId)
+		if err == nil {
+			connectorIdToName[id] = connector.Info.GetName()
+		}
 		return err
```

Comment:
A bit more verbose, but this is the more idiomatic way to write this.
```suggestion
		if err != nil {
			return err
		}
		connectorIdToName[id] = connector.Info.GetName()
		
		return nil
```

Reply:
(In the current approach, the fallthrough combined with the assumption that `err == nil` is confusing)